core/src/main/resources/hudson/PluginManager/_updateSite.js

```diff
@@ -0,0 +1,23 @@
+(function () {
+  showResetToDefaultButtonOrNot();
+})();
+
+function showResetToDefaultButtonOrNot() {
+  const resetButton = document.getElementById("reset-to-default");
+  const siteUrlInput = document.getElementById("update-site-url");
+  if (siteUrlInput.value === "https://updates.jenkins.io/update-center.json") {
+    resetButton.style.display = "none";
+  } else {
+    resetButton.style.display = "";
+  }
+}
+
+document.getElementById("update-site-url").onchange =
+  showResetToDefaultButtonOrNot;
+
+document.getElementById("reset-to-default").onclick = function (event) {
+  event.preventDefault();
+  const siteUrlInput = document.getElementById("update-site-url");
+  siteUrlInput.value = "https://updates.jenkins.io/update-center.json";
+  siteUrlInput.dispatchEvent(new Event("change"));
```

Comment:
the UX isn't very great here.

If I have this:
<img width="404" alt="image" src="https://user-images.githubusercontent.com/21194782/193256907-06cce08c-3b42-4a1b-8250-129df3f2bced.png">

then I make a change
<img width="460" alt="image" src="https://user-images.githubusercontent.com/21194782/193256946-80024779-dd8f-4ed5-8187-a3844dad12f4.png">

I click the Submit button:
<img width="415" alt="image" src="https://user-images.githubusercontent.com/21194782/193257005-00d2663b-01ce-4188-8ce2-0b56ea6da482.png">

Instead of the button being submitted I get the `Reset to default` appearing and my change isn't saved.

I this this should only be shown on page load, and only disappear if it's clicked, doesn't need to be as 'smart' as it is

Reply:
@timja Thanks for the suggestion! Have a look at the latest committed changes.